pkg/adaptor/cloud/azure/provider.go

```diff
@@ -251,7 +252,10 @@ func (p *azureProvider) DeleteInstance(ctx context.Context, instanceID string) e
 		return fmt.Errorf("creating VM client: %w", err)
 	}
 
-	pollerResponse, err := vmClient.BeginDelete(ctx, p.serviceConfig.ResourceGroupName, instanceID, nil)
+	// instanceID in the form of /subscriptions/<subID>/resourceGroups/<resource_name>/providers/Microsoft.Compute/virtualMachines/<VM_Name>.
+	vmName := strings.SplitAfter(instanceID, "virtualMachines/")[1]
```

Comment:
It feels a bit unfortunate that we have to mangle strings like this. If the azure api doesn't accept an id, but a name as an input for delete operations. we maybe should treat that name as an identifier for the azure code (afaik within a sub + resource-group a name would be a unique identifier). 

In any case I think we might want to be extra diligent with this operation, afaik delete operation will not return fail on non-existing operations (at least that's the behaviour for `az vm delete`. Some ideas:

- cover a failure of string extraction + return an error, so there's at least a log
- use a regex to extract it to be sure we are mangling a proper string:
`^/subscriptions/$sub/resourceGroups/$rg/providers/Microsoft.Compute/virtualMachines/(.*)$`

Reply:
Make sense, updated